src/nvim/api/vim.c

```diff
@@ -1344,6 +1344,8 @@ Dictionary nvim_get_mode(void)
 
   PUT(rv, "mode", STRING_OBJ(cstr_as_string(modestr)));
   PUT(rv, "blocking", BOOLEAN_OBJ(blocked));
+  // :help getcmdwintype()
+  PUT(rv, "wintype", STRING_OBJ(cchar_to_string((char)cmdwin_type)));
```

Comment:
Do we want these to be grouped under "cmdline" label somehow? I assume we want `wintype` to later also indicate modal terminal or plugin-defined `modal()` functionality being active.

Reply:
@bfredl latest commit returns names like `cmdline_:` and `cmdline_/`. Is that what you were thinking?

I think I'd prefer if we stuck to the standard plan `:` and `/` names, and these are simply understood (and documented) to mean "cmdline window". Then we have new identifiers for new window-types. Normal windows are "" (empty string).

Reply:
> I think I'd prefer if we stuck to the standard plain : and / names, and these are simply understood (and documented) to mean "cmdline window".

But there is no standard that says that one-letter chars identify cmdline window as opposed to other modal windows. We are defining this standard here and now, there is exactly nothing we have to "stick" to. One char strings are only used to distinguish the type of the _cmdline_ in something you already know is a cmdline window, for `getcmdwintype()` that already has "cmdwin" in the name.

It would be useful for an async plugin to identify cmdline windows overall without caring which sort it is. Cmdline window overrides a bunch of behaviors consistently, regardless of what the `firstc` happens to be. Should they check if the string is one byte long? Do we want to rule out `firstc` from ever being a unicode char? 

Checking `wintype=="cmdwin"` is self-documenting, while `strlen(wintype)==1` looks fragile and obscure. The firstc can be just a separate field if it is needed, we do not have to invent new stringly typed composites like `"cmdline_/"`.